src/test/java/org/candlepin/auth/permissions/PermissionFactoryTest.java

```diff
@@ -0,0 +1,409 @@
+
+        TestPermissionBlueprint ownerHypervisorBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.OWNER_HYPERVISORS.name(), Access.ALL.name());
+
+        TestPermissionBlueprint manageActivationKeyBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.MANAGE_ACTIVATION_KEYS.name(), Access.ALL.name());
+
+        Set<TestPermissionBlueprint> blueprints = Set.of(ownerBP, ownerPoolBP, usernameConsumerBP, attachBP,
+            ownerHypervisorBP, manageActivationKeyBP);
+
+        TestRole roleInfo = new TestRole()
+            .setPermissions(blueprints);
+
+        TestUser userInfo = new TestUser(false, ownerInfo)
+            .setRoles(List.of(roleInfo));
+
+        roleInfo.setUsers(List.of(userInfo));
+
+        List<Permission> actual = new PermissionFactory(ownerCurator)
+            .createPermissions(userInfo);
+
+        assertThat(actual)
+            .isNotNull()
+            // PermissionType.MANAGE_ACTIVATION_KEYS includes two permissions:
+            // ActivationKeyPermission and OwnerActivationKeyPermission
+            .hasSize(blueprints.size() + 1);
+    }
+
+    @Test
+    public void testCreatePermissionsWithNoExistingOwner() {
+        String ownerKey = TestUtil.randomString();
+        TestOwner ownerInfo = new TestOwner(ownerKey);
+
+        TestPermissionBlueprint ownerBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.OWNER.name(), Access.ALL.name());
+
+        TestPermissionBlueprint ownerPoolBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.OWNER_POOLS.name(), Access.ALL.name());
+
+        TestPermissionBlueprint usernameConsumerBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.USERNAME_CONSUMERS.name(), Access.ALL.name());
+
+        TestPermissionBlueprint attachBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.ATTACH.name(), Access.ALL.name());
+
+        TestPermissionBlueprint ownerHypervisorBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.OWNER_HYPERVISORS.name(), Access.ALL.name());
+
+        TestPermissionBlueprint manageActivationKeyBP = new TestPermissionBlueprint(ownerInfo,
+            PermissionType.MANAGE_ACTIVATION_KEYS.name(), Access.ALL.name());
+
+        Set<TestPermissionBlueprint> blueprints = Set.of(ownerBP, ownerPoolBP, usernameConsumerBP, attachBP,
+            ownerHypervisorBP, manageActivationKeyBP);
+
+        TestRole roleInfo = new TestRole()
+            .setPermissions(blueprints);
+
+        TestUser userInfo = new TestUser(false, ownerInfo)
+            .setRoles(List.of(roleInfo));
+
+        roleInfo.setUsers(List.of(userInfo));
+
+        List<Permission> actual = new PermissionFactory(ownerCurator)
+            .createPermissions(userInfo);
+
+        assertThat(actual)
+            .isNotNull()
+            .isEmpty();
+    }
+
+    public class TestOwner implements OwnerInfo {
```

Comment:
What do we get out of these classes that we're not getting from the concrete implementations or mocks? I don't see any logic in any of the methods.

Reply:
The problem with using the concrete implementations is that the `PermissionFactory.OwnerResolver` will always return that instance of `Owner` and we won't be able to test the actual case of `PermissionFactory.OwnerResolver.resolve` returning null.

https://github.com/candlepin/candlepin/blob/main/src/main/java/org/candlepin/auth/permissions/PermissionFactory.java#L83-L86

Reply:
I'm not following. Can you point to one of the tests where things are broken when an existing concrete implementation is used? I glanced through them again and it's not popping out to me.

Reply:
If you look at `PermissionFactoryTest.testCreatePermissionsWithNoExistingOwner` and used the concrete implementations, so populated a `User` object with a `Role` that has a `PermissionBlueprint` object that then has an `Owner` object on it. Then called `permissionFactory.createPermissions` using that `User`, the PermissionFactory.OwnerResolver.resolve would return that `Owner` object that is on the `PermissionBlueprint` instead of checking the cache or the DB.

Reply:
So we wouldn't be actually testing the case of the Owner not existing in the cache or DB

Reply:
Updated!